Kernel/Devices/Storage/VirtIO/VirtIOBlockDevice.cpp

```diff
@@ -120,7 +120,10 @@ UNMAP_AFTER_INIT ErrorOr<void> VirtIOBlockDevice::initialize_virtio_resources()
     }));
     TRY(setup_queues(1)); // REQUESTQ
     finish_init();
-    return {};
+
+    // This device is not added to DeviceManagement, because we've bypassed try_create_device
+    // Fix it now
+    return static_cast<Kernel::Device*>(this)->after_inserting();
```

Comment:
I dont think the cast is needed, just `return after_inserting()` should suffice

Reply:
The cast is needed, because the virtual method is redeclared private in StorageDevice.

Reply:
Interesting, seems like a bug to me, but whatever